File: src/test/java/no/utgdev/diy/ex2/Ex2Test.java
package no.utgdev.diy.ex2;

import no.utgdev.diy.DIYStatic;
import org.junit.jupiter.api.Test;

import java.lang.reflect.Method;
import java.util.Map;
import java.util.Set;

import static java.util.Collections.emptyList;
import static java.util.Collections.emptyMap;
import static org.assertj.core.api.Assertions.assertThat;

class Ex2Test {
    @Test
    void instansiate() {
        Set<Method> methods = DIYStatic.scan(Ex2Config.class);

        Map<String, Object> objects = DIYStatic.instansiate(methods);

        assertThat(objects).hasSize(4);
        assertThat(objects.get("number")).isEqualTo(2.0);
        assertThat(objects.get("string")).isEqualTo("");
        assertThat(objects.get("list")).isEqualTo(emptyList());
        assertThat(objects.get("map")).isEqualTo(emptyMap());
    }
}

File: src/test/kotlin/no/utgdev/diy/ex2/Ex2Test.kt
package no.utgdev.diy.ex2

import no.utgdev.diy.BeanDefinition
import no.utgdev.diy.DIYStatic
import no.utgdev.diy.DIYStatic.instansiate
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

internal class Ex2Test {
    @Test
    fun instansiate() {
        val methods: Set<BeanDefinition> = DIYStatic.scan(Ex2Config::class.java)
        val objects = instansiate(methods)
        Assertions.assertThat(objects).hasSize(4)
        Assertions.assertThat(objects["number"]).isEqualTo(2.0)
        Assertions.assertThat(objects["string"]).isEqualTo("")
        Assertions.assertThat(objects["list"]).isEqualTo(emptyList<Any>())
        Assertions.assertThat(objects["map"]).isEqualTo(emptyMap<Any, Any>())
    }
}